### PR TITLE
Handling rate-limited api calls

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -469,6 +468,12 @@ export class StreamChat {
 
 	doAxiosRequest = async (type, url, data, params = {}) => {
 		await this.tokenManager.tokenReady();
+
+		if (this.rateLimitResetTime) {
+			await sleep(this.rateLimitResetTime * 1000 - new Date());
```
Comment (ferhatelmas) [inline src/client.js:473]: might be better to add small noise so that all doesn't come at the same time

Comment (vishalnarkhede) [reply src/client.js:473]: noise?

Comment (ferhatelmas) [reply src/client.js:473]: jitter/delay - some small random time

Comment (vishalnarkhede) [reply src/client.js:473]: Makes sense!! Will do :)

**Pull request closed without merging**